packages/web-app-files/src/mixins/spaces/actions/setReadme.js

```diff
@@ -48,26 +47,26 @@ export default {
     }
   },
   methods: {
-    ...mapMutations('Files', ['UPDATE_RESOURCE_FIELD']),
+    ...mapMutations('Files', ['UPDATE_SPACE_FIELD']),
     ...mapActions(['showMessage']),
     async $_setSpaceReadme_trigger({ resources }) {
-      const space = this.currentFolder
-
       try {
+        const storageId = this.$route.params.storageId
+        const space = this.spaces.find((s) => s.id === storageId)
+
         const fileContent = await this.$client.files.getFileContents(resources[0].webDavPath)
         const fileMetaData = await this.$client.files.putFileContents(
           `/spaces/${space.id}/.space/readme.md`,
           fileContent
         )
-        this.UPDATE_RESOURCE_FIELD({
+        this.UPDATE_SPACE_FIELD({
```

Comment:
I think you need to do both here. The space can also be the current folder if you're setting the readme from within the project root folder.

Reply:
This should be sufficient as far as I can see. The README data gets fetched from the space in `state.spaces`, which gets updated via `UPDATE_SPACE_FIELD `.

Reply:
ok 👍